SJSU-1073_1 <NEW>Added pop-up alerts on the search widget

yuta05 opened the pull request:
Created a new branch from dev-general, SJSU-1073_1.

minhphan156 commented:
![image](https://user-images.githubusercontent.com/14304263/56396245-c8c75f80-61f2-11e9-8943-f142172036c7.png)

SandroSal commented on front_end/src/components/landing_page/search_widget/SearchWidget.js:
```diff
@@ -78,6 +82,27 @@ class SearchWidget extends Component {
+      this.state.checkOut !== ""
+    ) {
+      this.props.history.push("/searchResultOverview");
+    }
```
saveQuery and submitQuery should also be within the if statement, otherwise we call the backend with incomplete data.

SandroSal commented on front_end/src/components/landing_page/search_widget/SearchWidget.js:
```diff
@@ -78,6 +82,27 @@ class SearchWidget extends Component {
+      this.setState({ showDesAlert: false });
+    }
+    //Check if the checkIn and checkOut date are given or not.
+    if (this.state.checkIn === "" || this.state.checkOut === "") {
```
checkIn and checkOut can be null if a user enters a date and then manually deletes it. Please also check for null.

SandroSal commented on front_end/src/components/landing_page/search_widget/SearchWidget.js:
```diff
@@ -78,6 +82,27 @@ class SearchWidget extends Component {
+    }
+
+    if (
+      this.state.destinationName !== "" &&
```
gotta check for null here too then.

OR

create a local variable within onSearchClick(), something like
validationError = false;

then you set the validationError to true if one of the inputs are missing... in line 98 you only need to check validationError, and not all the different things again.

SandroSal commented on front_end/src/components/landing_page/search_widget/SearchWidget.js:
```diff
@@ -147,6 +172,20 @@ class SearchWidget extends Component {
     } else {
       return (
         <div className={classes.root}>
+          <Alert
```
If only the date is missing, it shows the date-error above the destination input field.

yuta05 commented:
Updated the following:

- Added saveQuery and submitQuery under if statement in onSearchClick().

- Added the alert for the deals search widget 

-  Changed to check for both " " and null

yuta05 commented:
Updated:

- error message stays over the Calendar input

- updated the error messages for mobile
